### DISCUSSION TITLE
Support for TestFlight (iOS) Internal Groups

### DISCUSSION THREAD

**Discussion by syewchuk** [fastlane/fastlane#19682] [Ideas] [answered]
TestFlight (iOS) added the concept of Internal Beta Groups at some point this year. (Previously there were only External Beta Groups) I would like to be able to use Fastlane to send builds to the internal groups. The key difference is that to send a build to an external groups it needs to be reviewed and approved first, whereas to the internal group it does not need to go through the review process.

I managed to use Fastlane to send a build to my internal group, BUT it also submitted the build for review. I do not want to submit the build for review, I just want to send it to the group.

I looked at the code in build_manager.rb and it looks like it ALWAYS first submits the build for review, and then adds the beta groups. Would it be possible to create an option to only send the build to the group (and not change the status to review)?

I suspect the upload to the internal groups would work as is if the build was not first set to review, but further investigation would be required.

Relevant code from build_manager.rb:

``` 
      if options[:groups] || options[:distribute_external]
        if uploaded_build.ready_for_beta_submission?
          uploaded_build.post_beta_app_review_submission
        else
          UI.message("Build #{uploaded_build.app_version} - #{uploaded_build.version} already submitted for review")
        end
      end

      if options[:groups]
        app = uploaded_build.app
        beta_groups = app.get_beta_groups.select do |group|
          options[:groups].include?(group.name)
        end

        unless beta_groups.empty?
          uploaded_build.add_beta_groups(beta_groups: beta_groups)
        end
      end
```

**Comment by horvathtamasattila** [comment, marked as answer]
I've just opened a [PR](https://github.com/fastlane/fastlane/pull/19735) about this with a possible solution, let's see how it goes.

**Reply by syewchuk** [reply]
Great, thanks for looking into this!
